android_jni/avifandroidjni/src/main/jni/libavif_jni.cc

```diff
@@ -124,9 +127,15 @@ FUNC(jboolean, decode, jobject encoded, int length, jobject bitmap) {
         decoder.decoder->image->height);
     return false;
   }
-  // Ensure that the bitmap format is RGBA_8888. For now, we only support that.
-  if (bitmap_info.format != ANDROID_BITMAP_FORMAT_RGBA_8888) {
-    LOGE("Bitmap format (%d) is not supported.", bitmap_info.format);
+  // Ensure that the bitmap format is either RGBA_8888 or RGBA_F16 based on the
+  // image's bitdepth.
+  if ((decoder.decoder->image->depth == 8 &&
+       bitmap_info.format != ANDROID_BITMAP_FORMAT_RGBA_8888) ||
```

Comment:
libavif can convert 8-bit YUVA to RGBA_F16. Would you like to support that?

I.e, we can simply check:

```
  // Ensure that the bitmap format is either RGBA_8888 or RGBA_F16.
  if (bitmap_info.format != ANDROID_BITMAP_FORMAT_RGBA_8888 &&
      bitmap_info.format != ANDROID_BITMAP_FORMAT_RGBA_F16) {
```

Reply:
I considered that feature. But F16 bitmaps and displaying HDR content in android are generally considered expensive and allowing this would indirectly encourage bad practice. So i thought it was best to keep this out.

Please let me know if you feel otherwise and i can update this.

Reply:
Whether an image is WCG or HDR depends on the image's ColorPrimaries and TranferCharacteristic value, so if F16 is required to display WCG/HDR content then maybe we shouldn't limit it there. Storing HDR in 8bit is probably bad, but it's not the fault of the image viewer (or alike).

Reply:
This issue came up in Chrome recently: https://crbug.com/1266456. As I understand it, Chrome renders HDR images in F16 image frame buffers, and 8-bit HDR images can't be rendered because the blink::AVIFImageDecoder class in Chrome does not have the code to convert 8-bit YUVA to F16 RGBA.

Your Android JNI bindings can easily support this if someone requests it.

Reply:
Sounds good. I have updated the PR to allow 8 -> 16 conversion as well.